hazelcast-sql-core/src/test/java/com/hazelcast/sql/SqlOrderByTest.java

```diff
@@ -0,0 +1,526 @@
+import com.hazelcast.config.IndexConfig;
+import com.hazelcast.config.IndexType;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.map.IMap;
+import com.hazelcast.sql.impl.SqlTestSupport;
+import com.hazelcast.test.HazelcastSerialParametersRunnerFactory;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import static com.hazelcast.sql.SqlBasicTest.AbstractPojo;
+import static com.hazelcast.sql.SqlBasicTest.AbstractPojoKey;
+import static com.hazelcast.sql.SqlBasicTest.DataSerializablePojo;
+import static com.hazelcast.sql.SqlBasicTest.DataSerializablePojoKey;
+import static com.hazelcast.sql.SqlBasicTest.IdentifiedDataSerializablePojo;
+import static com.hazelcast.sql.SqlBasicTest.IdentifiedDataSerializablePojoKey;
+import static com.hazelcast.sql.SqlBasicTest.PortablePojo;
+import static com.hazelcast.sql.SqlBasicTest.PortablePojoKey;
+import static com.hazelcast.sql.SqlBasicTest.SerializablePojo;
+import static com.hazelcast.sql.SqlBasicTest.SerializablePojoKey;
+import static com.hazelcast.sql.SqlBasicTest.SerializationMode;
+import static com.hazelcast.sql.SqlBasicTest.portableFieldName;
+import static com.hazelcast.sql.SqlBasicTest.serializationConfig;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * Test that covers basic column read operations through SQL.
+ */
+@RunWith(Parameterized.class)
+@Parameterized.UseParametersRunnerFactory(HazelcastSerialParametersRunnerFactory.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+@SuppressWarnings("checkstyle:RedundantModifier")
+public class SqlOrderByTest extends SqlTestSupport {
+
+    private static final String MAP_OBJECT = "map_object";
+    private static final String MAP_BINARY = "map_binary";
+
+    private static final int[] PAGE_SIZES = {256};
+    private static final int[] DATA_SET_SIZES = {4096};
+    private static final SqlTestInstanceFactory FACTORY = SqlTestInstanceFactory.create();
+
+    private static TestHazelcastInstanceFactory factory;
+    private static List<HazelcastInstance> members;
+    private static HazelcastInstance member;
+
+    private static HazelcastInstance member1;
+    private static HazelcastInstance member2;
+
+    @Parameter
+    public int cursorBufferSize;
```

Comment:
Do we really need to run the tests below with all the combinations of parameters?

Reply:
Simplified in https://github.com/hazelcast/hazelcast/pull/18013/commits/35921755279e215f77b86646b111e8e4810ce5b3